dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/ProgramStageWorkingListObjectBundleHook.java

```diff
@@ -0,0 +1,186 @@
+        validateDateFilterPeriod( errorReports, "EnrollmentCreatedDate", queryCriteria.getEnrolledAt() );
+        validateDateFilterPeriod( errorReports, "EnrollmentIncidentDate", queryCriteria.getEnrollmentOccurredAt() );
+        validateDateFilterPeriod( errorReports, "EventDate", queryCriteria.getEventCreatedAt() );
+        validateDateFilterPeriod( errorReports, "EventScheduledDate", queryCriteria.getScheduledAt() );
+
+        validateAssignedUsers( errorReports, queryCriteria.getAssignedUsers(), queryCriteria.getAssignedUserMode() );
+        validateOrganisationUnit( errorReports, queryCriteria.getOrgUnit(), queryCriteria.getOuMode() );
+        validateAttributeValueFilters( errorReports, queryCriteria.getAttributeValueFilters(),
+            teaService::getTrackedEntityAttribute );
+        validateDataFilters( errorReports, queryCriteria.getDataFilters() );
+
+        errorReports.forEach( addReports );
+    }
+
+    private void validateDateFilterPeriod( List<ErrorReport> errorReports, String item,
+        DateFilterPeriod dateFilterPeriod )
+    {
+        if ( dateFilterPeriod != null && dateFilterPeriod.getType() != null
+            && dateFilterPeriod.getType() == DatePeriodType.ABSOLUTE
+            && dateFilterPeriod.getStartDate() == null && dateFilterPeriod.getEndDate() == null )
+        {
+            errorReports.add( new ErrorReport( ProgramStageWorkingList.class, ErrorCode.E4062, item ) );
+        }
+    }
+
+    private void validateAssignedUsers( List<ErrorReport> errorReports, Set<String> assignedUsers,
+        AssignedUserSelectionMode userMode )
+    {
+        if ( CollectionUtils.isEmpty( assignedUsers )
+            && userMode == AssignedUserSelectionMode.PROVIDED )
+        {
+            errorReports.add( new ErrorReport( ProgramStageWorkingList.class, ErrorCode.E4063 ) );
+        }
+    }
+
+    private void validateOrganisationUnit( List<ErrorReport> errorReports, String orgUnit,
+        OrganisationUnitSelectionMode ouMode )
+    {
+        if ( orgUnit != null )
+        {
+            OrganisationUnit ou = organisationUnitService.getOrganisationUnit( orgUnit );
+            if ( ou == null )
+            {
+                errorReports
+                    .add( new ErrorReport( ProgramStageWorkingList.class, ErrorCode.E7500, orgUnit ) );
+                return;
+            }
+        }
+
+        if ( StringUtils.isEmpty( orgUnit )
+            && (ouMode == OrganisationUnitSelectionMode.SELECTED
+                || ouMode == OrganisationUnitSelectionMode.DESCENDANTS
+                || ouMode == OrganisationUnitSelectionMode.CHILDREN) )
+        {
+            errorReports.add( new ErrorReport( ProgramStageWorkingList.class, ErrorCode.E4064, ouMode ) );
+        }
+    }
+
+    private void validateDataFilters( List<ErrorReport> errorReports, List<EventDataFilter> dataFilters )
+    {
+        if ( !CollectionUtils.isEmpty( dataFilters ) )
+        {
+            dataFilters.forEach( avf -> {
```

Comment:
What does `avf` stand for? Wondering if that is a typo or intentional. EventDataFilter -> `evf`? Just `f` is enough as well I think 😄